src/Exception/HttpException.php

```diff
@@ -0,0 +1,212 @@
+<?php
+/**
+ * HTTP exception.
+ *
+ * HTTP exceptions are public-facing, so the messages are translated and should
+ * not reveal any sensitive information. Additional information can be attached
+ * for debugging.
+ *
+ * @package SatisPress
+ * @license GPL-2.0-or-later
+ * @since 0.3.0
+ */
+
+declare ( strict_types = 1 );
+
+namespace SatisPress\Exception;
+
+use SatisPress\Package;
+use SatisPress\Release;
+use Throwable;
+use WP_Http as HTTP;
+
+/**
+ * HTTP exception class.
+ *
+ * @since 0.3.0
+ */
+class HttpException extends \Exception implements ExceptionInterface {
+	/**
+	 * Exception data.
+	 *
+	 * @var array
+	 */
+	protected $data = [];
+
+	/**
+	 * HTTP status code.
+	 *
+	 * @var integer
+	 */
+	protected $status_code;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @param string    $message     Message.
+	 * @param integer   $status_code Optional. HTTP status code. Defaults to 500.
+	 * @param array     $data        Optional. Additional data.
+	 * @param integer   $code        Exception code.
+	 * @param Throwable $previous    Previous exception.
+	 */
+	public function __construct(
+		string $message,
+		int $status_code = HTTP::INTERNAL_SERVER_ERROR,
+		array $data = null,
+		int $code = 0,
+		Throwable $previous = null
+	) {
+		$this->data        = $data;
+		$this->status_code = $status_code;
+		$message           = $message ?: esc_html( 'Internal Server Error', 'satispress' );
```

Comment:
You've added a text domain, but the string is not internationalized.

Aside: I'm not convinced that Exception messages should be internationalized anyway.